cla_frontend/settings/base.py

```diff
@@ -179,21 +179,49 @@ def sentry_traces_sampler(sampling_context):
     ),
 )
 
-MIDDLEWARE_CLASSES = (
-    "core.middleware.MaintenanceModeMiddleware",
-    "django_statsd.middleware.GraphiteRequestTimingMiddleware",
-    "django_statsd.middleware.GraphiteMiddleware",
-    "django.contrib.sessions.middleware.SessionMiddleware",
-    "django.middleware.common.CommonMiddleware",
-    "django.middleware.csrf.CsrfViewMiddleware",
-    "django.contrib.auth.middleware.AuthenticationMiddleware",
-    "core.session_security.middleware.SessionSecurityMiddleware",
-    "django.contrib.messages.middleware.MessageMiddleware",
-    "django.middleware.clickjacking.XFrameOptionsMiddleware",
-    "cla_auth.middleware.ZoneMiddleware",
-    "core.middleware.Cla401Middleware",
-    "csp.middleware.CSPMiddleware",
-)
+DISABLE_SAMESITE_MIDDLEWARE = os.environ.get("DISABLE_SAMESITE_MIDDLEWARE", False)
+
+if not DISABLE_SAMESITE_MIDDLEWARE:
+    MIDDLEWARE_CLASSES = (
+        "core.middleware.MaintenanceModeMiddleware",
+        "django_statsd.middleware.GraphiteRequestTimingMiddleware",
+        "django_statsd.middleware.GraphiteMiddleware",
+        "django.contrib.sessions.middleware.SessionMiddleware",
+        "django.middleware.common.CommonMiddleware",
+        "django.middleware.csrf.CsrfViewMiddleware",
+        "django.contrib.auth.middleware.AuthenticationMiddleware",
+        "core.session_security.middleware.SessionSecurityMiddleware",
+        "django.contrib.messages.middleware.MessageMiddleware",
+        "django.middleware.clickjacking.XFrameOptionsMiddleware",
+        "cla_auth.middleware.ZoneMiddleware",
+        "core.middleware.Cla401Middleware",
+        "csp.middleware.CSPMiddleware",
+        "django_cookies_samesite.middleware.CookiesSameSite",
+        "djangosecure.middleware.SecurityMiddleware",
+    )
+else:
+    MIDDLEWARE_CLASSES = (
```

Comment:
Would it be possible to refactor this so you are not defining most of the middleware twice?

Could you add `"django_cookies_samesite.middleware.CookiesSameSite"` to the `MIDDLEWARE` afterwards, depending on the value of `DISABLE_SAMESITE_MIDDLEWARE`?  

I understand the order is important, however, there may be a better way to approach this without repeating yourself.